packages/babel-parser/src/plugins/typescript/index.ts

```diff
@@ -566,6 +566,16 @@ export default (superClass: ClassWithMixin<typeof Parser, IJSXParserMixin>) =>
 
       // For compatibility to estree we cannot call parseLiteral directly here
       node.argument = super.parseExprAtom() as N.StringLiteral;
+      if (this.hasPlugin("importAttributes")) {
+        node.options = null;
+      }
+      if (this.eat(tt.comma)) {
+        this.expectImportAttributesPlugin();
+        if (!this.match(tt.parenR)) {
+          node.options = super.parseMaybeAssignAllowIn();
```

Comment:
I implemented the same logic as `parseImportCall` ( https://github.com/babel/babel/blob/19c54cf690afe9d62e1dba8e6ee7d92bbe43451f/packages/babel-parser/src/parser/expression.ts#L2952-L2973 ). But I'm not sure what we should actually allow here.

Reply:
That looks good to me. We can add a new test case for integration with the JSX plugin, e.g.
```tsx
typeof import("foo.json", <ImportOptionBagJSX type="json"/>)
```

Reply:
I've added tests but it is invalid ( https://github.com/babel/babel/pull/16277/commits/950a377f8155f0b1421da9c8bc83dd68b416de87 )

Reply:
Hmm, `import("foo.json", <ImportOptionBagJSX type="json"/>)` works when we don't have `typescript` plugin. It is probably a parser bug.

Reply:
Should we fix it in this PR?

Reply:
Not necessary. The input is a purely imaginative case so I think we can defer the fix to the future.